store/client.go

```diff
@@ -19,21 +19,11 @@ type PCloudClient struct {
 	Token  string
 }
 
-// authResponse; internal representation of the auth response. Will be used to Unmarshal the response
-type authResponse struct {
-	Auth string
-}
-
 // uploadFileResponse; internal representation of the upload file response.
 type uploadFileResponse struct {
```

Comment:
I couldn't get rid of this struct, though. Couldn't make it work without it, not sure why, but I couldn't return the single int from the `Fileids` attribute using the `jsonResp := make(map[string]interface{})`

Reply:
I don't think those structs are unnecessary, though. I believe having strong typing at the unmarshalling time leads to more clear errors. For instance, now on line:

```go
return jsonResponse["auth"].(string), nil
```

You will get a cryptic runtime error instead of something along the lines of "I couldn't transform the JSON in struct foo".

What do you think?